virttest/test_setup.py

```diff
@@ -97,6 +100,90 @@ class PolkitConfigCleanupError(PolkitConfigError):
     pass
 
 
+class Setuper(object):
+
+    """
+    Virtual base abstraction of setuper.
+    """
+
+    __metaclass__ = ABCMeta
+
+    def __init__(self, test, params, env):
+        """
+        Initializes the setuper.
+
+        :param test: VirtTest instance.
+        :param params: Dictionary with the test parameters.
+        :param env: Dictionary with test environment.
+        """
+        self.test = test
+        self.params = params
+        self.env = env
+
+    @abstractmethod
+    def setup(self):
+        """Setup procedure."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def cleanup(self):
+        """Cleanup procedure."""
+        raise NotImplementedError
+
+
+class SetuperMgr(object):
+
+    """
+    Setuper Manager implementation.
+
+    The instance can help do the setup stuff before test started and
+    do the cleanup stuff after test finished. This setup-cleanup
+    combined stuff will be performed in LIFO order.
+    """
+
+    def __init__(self, test, params, env):
+        """
+        Initializes the setuper manager.
+
+        :param test: VirtTest instance.
+        :param params: Dictionary with the test parameters.
+        :param env: Dictionary with test environment.
+        """
+        self._stp_args = (test, params, env)
```

Comment:
unnecessary shortcuts here, it'd improve readability if you'd used `_setup_args`

Reply:
OK.